problem with generic esp8266

C:\Program Files (x86)\Arduino\libraries\GPS-neo-6m\Gpsneo.cpp:4:27: fatal error: avr/pgmspace.h: No such file or directory

 #include <avr/pgmspace.h> 

                           ^

compilation terminated.

Multiple libraries were found for "SoftwareSerial.h"
 Used: C:\Users\ANAND\AppData\Local\Arduino15\packages\esp8266\hardware\esp8266\2.4.0\libraries\SoftwareSerial
 Not used: C:\Program Files (x86)\Arduino\libraries\SoftwareSerial
Using library SoftwareSerial at version 1.0 in folder: C:\Users\ANAND\AppData\Local\Arduino15\packages\esp8266\hardware\esp8266\2.4.0\libraries\SoftwareSerial 
Using library GPS-neo-6m in folder: C:\Program Files (x86)\Arduino\libraries\GPS-neo-6m (legacy)
exit status 1
Error compiling for board Generic ESP8266 Module.

plz to help me to sort it out
Thanks in advance

--- a/Gpsneo.cpp
+++ b/Gpsneo.cpp
@@ -1,7 +1,11 @@
 
 #include "Arduino.h"
 #include "Gpsneo.h"
-#include <avr/pgmspace.h> 
+#ifdef ESP8266
+#include <pgmspace.h>
+#else //ESP8266
+#include <avr/pgmspace.h>
+#endif  //ESP8266
 
 
 const char string_0[] PROGMEM = "http://www.google.com/maps/place/";

